src/helpers/lib/methodRegistry.js

```diff
@@ -33,14 +14,16 @@ const REGISTRY_MAP = {
 
 export default class MethodRegistry {
   constructor (opts = {}) {
-    this.eth = opts.eth || new Eth(DEFAULT_ETH_NODE)
+    this.provider =
+      opts.provider || new ethers.providers.WebSocketProvider(DEFAULT_ETH_NODE)
     this.network = opts.network || '1'
   }
 
-  // !!! This function can mutate `this.eth`
+  // !!! This function can mutate `this.provider`
   async initRegistry () {
-    if (await this.eth.net.getId() !== '1') {
-      this.eth = new Eth(DEFAULT_ETH_NODE)
+    const network = await this.provider.getNetwork()
+    if (network.chainId !== 1) {
+      this.provider = new ethers.providers.WebSocketProvider(DEFAULT_ETH_NODE)
```

Comment:
This is the part I'm worried most about.

As discussed offline, let's evaluate some ways to make this method look up a bit more modular (and with additional power, e.g. checking 4bytes)!